app/src/components/trace/TraceTree.tsx

```diff
@@ -68,7 +171,17 @@ function SpanTreeItem<TSpan extends ISpanItem>(props: {
   const { node, selectedSpanNodeId, onSpanClick, nestingLevel = 0 } = props;
   const childNodes = node.children;
   const [isCollapsed, setIsCollapsed] = useState(false);
+  const { isCollapsed: treeIsCollapsed } = useTraceTree();
   const hasChildren = childNodes.length > 0;
+  const showMetricsInTraceTree = usePreferencesContext(
+    (state) => state.showMetricsInTraceTree
+  );
+
+  // React to global changes to the trace tree state and change local state
+  useEffect(() => {
+    setIsCollapsed(treeIsCollapsed);
+  }, [treeIsCollapsed]);
```

Comment:
whats the rationale behind this vs just using the treeIsCollapsed directly?

Reply:
This lets local state take precedence over the global so you can expand as you go